plan/logical_plan_builder.go

```diff
@@ -1800,10 +1812,20 @@ func (b *planBuilder) buildSemiJoin(outerPlan, innerPlan LogicalPlan, onConditio
 	if b.TableHints() != nil {
 		outerAlias := extractTableAlias(outerPlan)
 		innerAlias := extractTableAlias(innerPlan)
-		joinPlan.preferMergeJoin = b.TableHints().ifPreferMergeJoin(outerAlias, innerAlias)
+		if b.TableHints().ifPreferMergeJoin(outerAlias, innerAlias) {
+			joinPlan.preferJoinType |= preferMergeJoin
+		}
+		if b.TableHints().ifPreferHashJoin(outerAlias, innerAlias) {
+			joinPlan.preferJoinType |= preferHashJoin
+		}
 		// semi join's outer is always the left side.
 		if b.TableHints().ifPreferINLJ(outerAlias) {
-			joinPlan.preferINLJ = preferLeftAsOuter
+			joinPlan.preferJoinType = preferLeftAsIndexOuter
+		}
+		// If there're multiple join hints, they're conflict.
+		if bits.OnesCount(joinPlan.preferJoinType) > 1 {
+			b.err = errors.New("Optimizer Hints is conflict")
```

Comment:
make error message more specific ? like: Optimizer Hints conflict, at most one of {TIDB_INLJ, TIDB_SMJ, TIDB_HASHJOIN} join hints should be specified.